### PR TITLE
golanci-lint based cleanup

### PR DESCRIPTION
**Pull request by iwankgb**
Is something like this useful? Majority of fixes is about not checked errors (in and out of tests). After applying more complex ones it should be possible to use `golangci-lint` and I think it could benefit the project.

### CODE REVIEW

File: utils/oomparser/oomparser.go [hunk 1]
```diff
@@ -107,11 +107,8 @@ func getProcessNamePid(line string, currentOomInstance *OomInstance) (bool, erro
 
 // uses regex to see if line is the start of a kernel oom log
 func checkIfStartOfOomMessages(line string) bool {
-	potential_oom_start := firstLineRegexp.MatchString(line)
-	if potential_oom_start {
-		return true
-	}
-	return false
+	potentialOomStart := firstLineRegexp.MatchString(line)
```
Comment (dashpole) [inline utils/oomparser/oomparser.go:110]: just `return firstLineRegexp.MatchString(line)`?

**Comment by iwankgb** [comment]
`golint` is one of supported linters. Besides this it is capable of detecting unchecked errors, deadcode, unused properties/functions/variables/struts and many more (it looks like about 30 linters are supported).
I was thinking about using default set of linters for starters and idea of adding `golint` to this set sounds really good. I don't think it makes sense to look for more at the moment as some tuning will be required and the community should choose what is important and what is not.

**Comment by dashpole** [comment]
That sounds reasonable.

**Comment by iwankgb** [comment]
At the moment some check from default suite fail - fixing them seem to be non-trivial and I wanted to keep this PR as uncomplicated as possible.

**Comment by dashpole** [comment]
Ok, lets iterate then.

**Pull request merged**